boa/src/builtins/array/mod.rs

```diff
@@ -48,16 +48,13 @@ impl Array {
                 .expect("Could not get global object"),
         ));
         array.set_data(ObjectData::Array);
-        array.borrow().set_internal_slot(
-            INSTANCE_PROTOTYPE,
-            interpreter
-                .realm()
-                .environment
-                .get_binding_value("Array")
-                .expect("Array was not initialized")
-                .borrow()
-                .get_field(PROTOTYPE),
-        );
+        array.as_object_mut().unwrap().prototype = interpreter
+            .realm()
+            .environment
+            .get_binding_value("Array")
+            .expect("Array was not initialized")
+            .borrow()
+            .get_field(PROTOTYPE);
```

Comment:
We could have a `prototype()` getter here, and the `get_field()` use that function if the passed string is the correct one. I think this would make it faster most of the time.

Reply:
I think we could store the builtin objects like `Array`, `String`, `Number` etc, in the interpreter or Realm as fields as a `GcObject` this will eliminate the Object lookup an we could do something like:
```rust
interperter.realm().array_object().prototype()
``` 
So we would cache the them.